sdk/compiler/damlc/daml-doc/src/DA/Daml/Doc/Extract.hs

```diff
@@ -123,18 +129,30 @@ collectDocs ds
             then Nothing
             else Just . DocText . T.strip $ T.unlines docs
 
-    getNextOrPrevDoc :: LHsDecl a -> Maybe DocText
-    getNextOrPrevDoc = \case
-        L _ (DocD _ (DocCommentNext str)) -> Just (docToText str)
-        L _ (DocD _ (DocCommentPrev str)) -> Just (docToText str)
-            -- technically this is a malformed doc, but we'll take it
+    getNextOrPrevOrWarnDoc :: LHsDecl a -> Maybe (Either DocText DocText)
```

Comment:
Since we don't extract WarnDoc so much as Warning declarations, can we resort to `getNextDocOrPrevDocOrWarnDec`?